code/modules/mob/living/simple_animal/friendly/beachcarp.dm

```diff
@@ -0,0 +1,65 @@
+/mob/living/simple_animal/beachcarp //im unsure if this is a good idea, ask mark or tmt
+	name = "carp"
+	desc = "A passive, fang-bearing creature that resembles a fish."
+	icon = 'icons/mob/fish.dmi'
+	icon_state = "base"
+	icon_living = "base"
+	icon_dead = "base_dead"
+	icon_gib = "carp_gib"
+	speak_emote = list("glubs")
+	emote_hear = list("glubs.")
+	emote_see = list("glubs.")
+	speak_chance = 1
+	faction = list("mining")
+	turns_per_move = 5
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/carpmeat = 2)
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "punches"
+	response_harm_simple = "punch"
+	stop_automated_movement = 1
+	friendly_verb_continuous = "nibbles"
+	friendly_verb_simple = "nibble"
+	turns_per_move = 2
+
+
+/mob/living/simple_animal/beachcarp/Life()
```

Comment:
is the only thing this is for, moving onto only water turfs? if yes, i am really not a fan of this solution